src/sentry/mediators/token_exchange/refresher.py

```diff
@@ -68,18 +83,19 @@ def token(self):
         try:
             return ApiToken.objects.get(refresh_token=self.refresh_token)
         except ApiToken.DoesNotExist:
-            raise APIUnauthorized
+            sentry_sdk.capture_message("Unable to find given refresh token in ApiToken objects")
```

Comment:
We could include the first/last 4 characters of the refresh token as event context so that we have a bit more context than now.